src/components/structures/TimelinePanel.tsx

```diff
@@ -619,6 +627,54 @@ class TimelinePanel extends React.Component<IProps, IState> {
         this.forceUpdate();
     };
 
+    // Called whenever the visibility of an event changes, as per
+    // MSC3531. We typically need to re-render the tile.
+    private onEventVisibilityChange = (ev: MatrixEvent): void => {
+        const roomId = ev.getRoomId();
+        if (this.unmounted) {
+            return;
+        }
+
+        // ignore events for other rooms
+        if (roomId !== this.props.timelineSet.room.roomId) {
```

Comment:
Some timeline set can be instanciated without a `room` attribute.
The one instance that comes to mind in the notifications sidebar (the bell icon on the top right of Element Web)